### DISCUSSION TITLE
help: get started quickly screen error - cannot use budibase

### DISCUSSION THREAD

**Discussion by dncpax** [Budibase/budibase#2997] [Help] [answered]
Hi there.

After installing budibase, I navigate to the home page, and get this error:
SyntaxError: Unexpected token u in JSON at position 0

So there are no buttons to continue - I cannot create an app nor import an app. So nothing works really.

The screen starts like this:
![image](https://user-images.githubusercontent.com/2836636/137290336-b79f9897-0d9f-4a00-ba07-a8f1699bdf86.png)

And ends up with this error:
![image](https://user-images.githubusercontent.com/2836636/137290431-0104badd-5074-4b05-b697-ccbd89e3b15b.png)

I have an upstream proxy, that I have tried to setup.
I have access to the internet from the envoyproxy container.

I couldn't figure out if this is true from the  bbapps container... I tried but couldn't figure out the shell... no bash, no apt, no ping...

So any help is welcome...

Thanks.

**Comment by mjashanks** [comment]
Hi @dncpax. I wasn't able to run `docker exec -it bbapps bash`, but `docker exec -it  bbapps ip addr` works - not sure if that can help you - this is a bit outside of my knowledge, but will try and get some help for you.

We shouldn't be depending on pulling templates from the internet all the same - I'll log an issue for that.

**Reply by dncpax** [reply]
Thanks @mjashanks.

I've located the error by looking at bbapps logs. It's failing requesting https://prod-budi-templates.s3-eu-west-1.amazonaws.com/manifest.json.

The full error is below.
Thanks.

```
ERROR [1634203650305] (29 on f4b3a55c28ed): request to https://prod-budi-templates.s3-eu-west-1.amazonaws.com/manifest.json failed, reason: connect ETIMEDOUT 52.218.20.43:443
    message: "request to https://prod-budi-templates.s3-eu-west-1.amazonaws.com/manifest.json failed, reason: connect ETIMEDOUT 52.218.20.43:443"
    type: "system"
    errno: "ETIMEDOUT"
    code: "ETIMEDOUT"
    stack: "FetchError: request to https://prod-budi-templates.s3-eu-west-1.amazonaws.com/manifest.json failed, reason: connect ETIMEDOUT 52.218.20.43:443\n    at ClientRequest.<anonymous> (/app/node_modules/node-fetch/lib/index.js:1455:11)\n    at ClientRequest.emit (events.js:326:22)\n    at ClientRequest.EventEmitter.emit (domain.js:483:12)\n    at TLSSocket.socketErrorListener (_http_client.js:427:9)\n    at TLSSocket.emit (events.js:314:20)\n    at TLSSocket.EventEmitter.emit (domain.js:483:12)\n    at emitErrorNT (internal/streams/destroy.js:92:8)\n    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)\n    at processTicksAndRejections (internal/process/task_queues.js:84:21)"
    req: {
      "id": 21,
      "method": "GET",
      "url": "/api/templates?type=app",
      "headers": {
        "host": "redacted:10000",
        "x-budibase-app-id": "",
        "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36",
        "content-type": "application/json",
        "accept": "*/*",
        "sec-gpc": "1",
        "referer": "http://redacted:10000/builder/portal/apps",
        "accept-encoding": "gzip, deflate",
        "accept-language": "pt-PT,pt;q=0.9,en-US;q=0.8,en;q=0.7",
        "cookie": "budibase:auth=redacted; budibase:auth.sig=redacted",
        "x-forwarded-proto": "http",
        "x-request-id": "b3440c5e-3832-4417-befb-c0f3b9619e2c",
        "x-envoy-expected-rq-timeout-ms": "15000",
        "content-length": "0"
      },
      "remoteAddress": "::ffff:172.18.0.9",
      "remotePort": 43450
    }
Trace: FetchError: request to https://prod-budi-templates.s3-eu-west-1.amazonaws.com/manifest.json failed, reason: connect ETIMEDOUT 52.218.20.43:443
    at ClientRequest.<anonymous> (/app/node_modules/node-fetch/lib/index.js:1455:11)
    at ClientRequest.emit (events.js:326:22)
    at ClientRequest.EventEmitter.emit (domain.js:483:12)
    at TLSSocket.socketErrorListener (_http_client.js:427:9)
    at TLSSocket.emit (events.js:314:20)
    at TLSSocket.EventEmitter.emit (domain.js:483:12)
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  type: 'system',
  errno: 'ETIMEDOUT',
  code: 'ETIMEDOUT'
}
    at /app/dist/api/index.js:69:21
    at Generator.throw (<anonymous>)
    at rejected (/app/dist/api/index.js:6:65)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

**Reply by mjashanks** [reply]
Thanks!

**Comment by dncpax** [comment, marked as answer]
I've found a workaround to get things going for now...

I host the manifest.json somewhere accessible to bbapps docker container, and override the dns lookup with a local hosts file.
To make this happen I add to the docker-compose.yaml a "extra_hosts" section (more info https://docs.docker.com/compose/compose-file/compose-file-v3/).
So in the part regarding bbapps I added:
```
extra_hosts:
      - "prod-budi-templates.s3-eu-west-1.amazonaws.com:<local_ip>"
```

Then docker-compose  --env-file hosting.properties up.
After this, bbapps has a /etc/hosts file with this in it:
```
127.0.0.1       localhost
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
<local_ip>    prod-budi-templates.s3-eu-west-1.amazonaws.com
172.19.0.8      6a9aadcf84c3
```

As long as there's a file being hosted at `https://<local_ip>/manifest.json` everything works.

Now everytime bb updates, I have to repeat the same thing since the docker-compose.yaml gets overwritten...

Hope there's a better solution.

Thanks.

**Reply by mjashanks** [reply]
Ah nice one. We have just released a self-hosting update, in the last hour.

On the next release, we will have this issue fixed - i.e. the error won't appear, you just won't see any templates.
